apps/prairielearn/src/pages/instructorAssessmentQuestions/editQuestionModal.html.ts

```diff
@@ -0,0 +1,184 @@
+import { html } from '@prairielearn/html';
+
+import { Modal } from '../../components/Modal.html.js';
+
+export function EditQuestionModal({
+  newQuestion,
+  question,
+  zoneIndex,
+  questionIndex,
+  alternativeIndex,
+  assessmentType,
+}: {
+  newQuestion: boolean;
+  question?: Record<string, any>;
+  urlPrefix?: string;
+  assessmentInstanceId?: string;
+  zoneIndex: number;
+  questionIndex: number;
+  alternativeIndex?: number;
+  assessmentType: string;
+}) {
+  return Modal({
+    id: 'editQuestionModal',
+    title: newQuestion ? 'Add Question' : 'Update Question',
+    body: html`
+      <input type="hidden" name="zoneIndex" id="zoneIndexInput" value="${zoneIndex}" />
+      <input type="hidden" name="questionIndex" id="questionIndexInput" value="${questionIndex}" />
+      <input
+        type="hidden"
+        name="alternativeIndex"
+        id="alternativeIndexInput"
+        value="${alternativeIndex}"
+      />
+      <input type="hidden" name="title" id="titleInput" value="${question?.title}" />
+      <input type="hidden" name="tags" id="tagsInput" value="${JSON.stringify(question?.tags)}" />
+      <input
+        type="hidden"
+        name="otherAssessments"
+        id="otherAssessmentsInput"
+        value="${JSON.stringify(question?.other_assessments)}"
+      />
+      <input
+        type="hidden"
+        name="topic"
+        id="topicInput"
+        value="${JSON.stringify(question?.topic)}"
+      />
+      <div class="form-group">
+        <label for="qid">QID</label>
+        <div class="input-group">
+          <input
+            type="text"
+            class="form-control qid"
+            id="qidInput"
+            name="qid"
+            aria-describedby="qidHelp"
+            value="${question?.qid}"
+          />
+          <div class="input-group-append">
+            <button type="button" class="btn btn-primary" id="findQid">
+              <i class="fa fa-magnifying-glass" aria-hidden="true"></i> Find QID
+            </button>
```

Comment:
Currently, typing in the QID in the input field does not fetch the question data as I am only doing this after using the 'find QID' modal and storing the `topics` and `tags` from that in hidden form inputs here. I can either:

1) disable the input and force users to use the 'Find QID' button when selecting a question. This would work for now, but seems less than ideal for any users that already know the QID they want to include. 
2) fetch data for all questions on page load. This would allow us to just grab the question data that we already have access to but might be unnecessary. 
3) fetch question data for this specific question after clicking the 'Update Question' button in this modal. This would require another endpoint but we could query the db for the specific qid and respond with just that question data. 

Let me know if any of these seem like the best method or if there is another solution here.

Reply:
I think (3) is the best option in the long run. In the interest of not further growing this PR, I'd probably lean towards (1) temporarily.

Had we talked about breaking up this task earlier, I would have advised only implementing "type in the QID" to begin with, and then layering on the question picker in future PRs.